include/boost/url/bnf/parse.hpp

```diff
@@ -20,54 +20,42 @@ namespace boost {
 namespace urls {
 namespace bnf {
 
-/*  Design
 
-    - An element is default constructible
-    - Holds the result of parsing
-    - overloads the free function `parse`
-    - has a nested value_type
-    - has `value_type& operator*() const noexcept`
-*/
-
-/** @brief Parse a literal character
+/** Parse a literal character
 
    This function parses the string defined by the first chars in the
-   range of chars [@ref it, @ref end) as a literal character.
-
-   The function returns `true` and resets @ref ec when the input is
-   parsed successfully. The initial character in the range is immediately
-   consumed and ignored.
+   range of chars [`it`, `end`) as a literal character.
 
-   If the input cannot be parsed into the literal BNF literal,
-   the function returns `false` and sets the error number in @ref ec to the
-   appropriate error number defined in @ref boost::urls::error.
+   The initial character in the range is immediately consumed and ignored.
```

Comment:
This is a little confusing because it implies that the character is consumed no matter what, which is incorrect.